### ISSUE TITLE
Struct fields expansion in other structs Assist

### ISSUE THREAD

**Issue by harudagondi** [rust-lang/rust-analyzer#15389]
```rust
struct Point2D {
    x: i32,
    y: i32,
}

struct Point3D {
    ..Point2D$0
    z: i32,
}
```

Expands to:

```rust
struct Point2D {
    x: i32,
    y: i32,
}

struct Point3D {
    x: i32,
    y: i32,
    z: i32,
}
```

**Comment by alibektas** [comment]
I was wondering what this assist should be doing in case of fields with lifetimes. Consider 

```rust
struct Source<'a , 'b> {
    a : &'a str,
    b : &'b str
}

struct Target<'a> {
    src : Source 
}
```

In a case like this we can theoretically put the only possible lifetime annotation `'a` after the transformation but in more complex examples it is almost inevitable that we are going to produce false positives. I have two questions regarding this

1. I don't see any way to pose comparative questions for lifetimes. For example I want to know 

```rust
struct ComplexStruct<'a, 'b, T>
where
    'a: 'b,
    T: 'static + Clone,
{
    field1: &'a str,
    field2: &'b T,
    field3: Vec<&'a str>,
}```

 how `'a` relates to `'b`. Are there any utils in HIR/MIR that does this?

2. Regardless of the answer to (1). Should I consider lifetimes or not?

**Comment by Veykril** [comment]
```
struct Source<'a , 'b> {
    a : &'a str,
    b : &'b str
}

struct Target<'a> {
    src : Source 
}
```
This code is incorrect anyways, you can't omit lifetimes in fields, so the correct code example would be
```rs
struct Source<'a , 'b> {
    a : &'a str,
    b : &'b str
}

struct Target<'a> {
    src : Source<'a, 'a>
}
```
in this case, there you can relate the lifetimes properly (and in fact, we never have to even think abou introducing new generic parameters because they have to be substitued prior to the assist already.
